pkg/bosh/manifest/manifest.go

```diff
@@ -158,6 +158,11 @@ type duplicateYamlValue struct {
 
 // LoadYAML returns a new BOSH deployment manifest from a yaml representation
 func LoadYAML(data []byte) (*Manifest, error) {
+	return LoadYAMLWithName(data, nil)
+}
+
+// LoadYAMLWithName returns a new BOSH deployment manifest from a yaml representation with the given name
+func LoadYAMLWithName(data []byte, name *string) (*Manifest, error) {
```

Comment:
This should be set as soon as possible, and should already be set in the desired manifest.
Do you need it _before_ the desired manifest is a thing?

Reply:
We started without this change. But it turned out that we got the wrong service names (DNS entries) to wait for.

Reply:
You can see it also in the changed test `resolver_test.go`, that the name of the loaded  manifest was not correct. 
It was quite confusing for us, that the name was not correct when the manifest is loaded the first time.

Reply:
Historically we assumed the bdpl name would be equal to the manifest name. I think we discussed to add a check for that, but must have somehow ended up with the enforcement in `deployment_reconciler.go`.

The behavior/expectation is also documented, at least in once place: https://github.com/cloudfoundry-incubator/cf-operator/blame/master/docs/from_bosh_to_kube.md#L60

That's also the problem with the [resolver_test.go](https://github.com/cloudfoundry-incubator/cf-operator/blob/5c7bb82f371627cf92516d321ab26c4b314b95f3/pkg/bosh/converter/resolver_test.go#L136), the test data is 'wrong'.

I'm not sure we have the best approach here, convention and partial enforcement. Maybe it makes sense to ignore the bdpl name and use the manifest name everywhere. Syncing these two values everywhere in the code seems brittle, as you discovered ;)

However, I'd rather have the  clean up in a proper story so we have a consistent approach to this design issue?

Reply:
> How about we always use the bdpl name, and we don't even [de]serialize the name?
> Then you **always** have to provide the name when you call LoadYaml, and you can error if it's empty.

You mean in a separate story/PR, right?

Reply:
Seems complicated to do, e.g. the `variable-interpolation` command takes a file name and doesn't know the name of the bdpl resource.

Reply:
Should this block this PR? Or is something else missing? It seems that this is not directly  related to this PR, right?

Reply:
@c0d1ngm0nk3y you added this function to the package, and I don't think it makes sense.
As a developer, I don't know why that package is exposing these 2 functions and what the differences are. We can create another story, but I thought my suggestion would be fast to implement.
If you don't think so, please let me know.

Reply:
@viovanov quoting @manno 
> Seems complicated to do, e.g. the variable-interpolation command takes a file name and doesn't know the name of the bdpl resource.

It seems not to be straight forward. I'd propose to merge this and create a chore to clean up the use of manifest and deployment names. WDYT?

Reply:
@viovanov I agree that a 2nd `LoadYaml` is confusing and for sure not what we want in the end. But I  think the existing `LoadYaml` is kind of buggy  since it returns different outcomes depending on when it is called. But `LoadYaml` is called quite often, so we didn't want to change this without  understanding the consequences in depth. I think it is an existing bug and not directly  related to this PR, but we can have a look. I was just a bit careful because of the "Seems complicated to do" @manno

Reply:
Fair enough - we could merge and schedule something else. This has been open for a while.

Reply:
https://www.pivotaltracker.com/story/show/169575453

Reply:
Yes, it's been a while and I am keen of getting this in. So, please let me know when you see something that should be addressed/clarified :)

@manno Could you perhaps clarify which change you would like to see to get this merged?

Reply:
@c0d1ngm0nk3y @loewenstein 
Please focus this PR on the update serial behavior, changing the `LoadYAML` is another discussion and should be a separate story.

I still think you should remove the new `LoadYAMLWithName` from this PR, if you fix the test data I [pointed to earlier](https://github.com/cloudfoundry-incubator/cf-operator/blob/5c7bb82f371627cf92516d321ab26c4b314b95f3/pkg/bosh/converter/resolver_test.go#L136).